File: dotnet-desktop-guide/net/winforms/controls-design/snippets/how-to-create-simple-custom-control/vb/DefaultTemplate/FirstControl.vb
Imports System.ComponentModel

Namespace DefaultTemplate

    ' <template>
    Public Class FirstControl

        Protected Overrides Sub OnPaint(ByVal e As System.Windows.Forms.PaintEventArgs)
            MyBase.OnPaint(e)

            'Add your custom paint code here
        End Sub

    End Class
    ' </template>

    Public Class SomethingElse : Inherits System.Windows.Forms.Control
        Private _textAlignment As HorizontalAlignment = HorizontalAlignment.Left


#Disable Warning WFO1000 ' Missing code serialization configuration for property content
        ' <property>
        Public Property TextAlignment As HorizontalAlignment
            Get
                Return _textAlignment
            End Get

            Set(value As HorizontalAlignment)
                _textAlignment = value
                Invalidate()
            End Set
        End Property
        ' </property>
#Enable Warning WFO1000 ' Missing code serialization configuration for property content
    End Class

End Namespace

File: dotnet-desktop-guide/net/winforms/controls-design/snippets/how-to-create-simple-custom-control/csharp/DefaultTemplate/FirstControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace UserControlProject.DefaultTemplate;

// <template>
public partial class FirstControl : Control
{
    public FirstControl()
    {
        InitializeComponent();
    }

    protected override void OnPaint(PaintEventArgs pe)
    {
        base.OnPaint(pe);
    }
}
// </template>

public class SomethingElse : Control
{
    private HorizontalAlignment _textAlignment = HorizontalAlignment.Left;

    [System.ComponentModel.Category("Alignment"),
    System.ComponentModel.Description("Specifies the alignment of text."),
    System.ComponentModel.DefaultValue(HorizontalAlignment.Left)]
    // <property>
    public HorizontalAlignment TextAlignment
    {
        get => _textAlignment;
        set
        {
            _textAlignment = value;
            Invalidate();
        }
    }
    // </property>
}
